### ISSUE TITLE
dexador does not work with mitmproxy (Invalid HTTP request form (expected: authority or absolute, got: relative))

### ISSUE THREAD

**Issue by Gira-X** [fukamachi/dexador#99]
This simple call fails in [mitmproxy](https://mitmproxy.org/) while it works in [Proxyman](https://proxyman.io/) without issues:

```lisp
(dex:get "http://lisp.org/" :proxy "http://localhost:9090/")
```

The debugger reports this error which is coming from `mitmproxy`, started with `mitmproxy -p 9090`:

```
An HTTP request to "http://lisp.org/" returned 400 bad request.

<html>
            <head>
                <title>400 Bad Request</title>
            </head>
            <body>
            <h1>400 Bad Request</h1>
            <p>HttpException(&#x27;Invalid HTTP request form (expected: authority or absolute, got: relative)&#x27;)</p>
            </body>
        </html>
```
  
With `:verbose t`, this is printed:

```
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
GET / HTTP/1.1
User-Agent: Dexador/0.9.14 (SBCL 2.1.5); Darwin; 18.7.0
Host: lisp.org
Accept: */*
Content-Length: 0

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
HTTP/1.1 400 Bad Request
Server: mitmproxy 6.0.2
Connection: close
Content-Length: 305
Content-Type: text/html

<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```

Any idea how to fix this? I suspect `mitmproxy` breaks on the `GET / HTTP/1.1` line because there is no full path?

**Comment by Gira-X** [comment]
Interestingly, `mitmproxy` logs the calls correctly when run in transparent mode via `(dex:get "http://lisp.org/")`, but it is a bit cumbersome to setup and has issues with Docker and XAMPP for me, so I prefer to not use the transparent mode.

**Comment by Gira-X** [comment]
I compared with `Drakma` and there `mitmproxy` works without problems:

```lisp
(drakma:http-request "http://lisp.org/" :proxy '("localhost" 9090))
```

Is it a bug in `dexador` or how proxies work internally with the port?

**Comment by ajberkley** [comment]
I setup mitmproxy (with no certs) to see if this is still the case and
(dex:get "http://lisp.org/" :proxy "http://localhost:9090/" :insecure t)
works for me (using the current github version of dexador).

Can you explain the setup you are using in more detail?

**Comment by Gira-X** [comment]
Aha, adding `:insecure t` works and the call is logged in mitmproxy.

Without it I get this error:

```bash
SSL verify error: 20 x509_v_err_unable_to_get_issuer_cert_locally
   [Condition of type cl+ssl:ssl-error-verify]
```

Now I see that`:insecure t` is required because http://lisp.org/ forwards via 301 to https://lisp.org/.

---

I tested a purely HTTP site like this: `(dex:get "http://t3x.org/index.html" :proxy "http://localhost:9090/")` (I just picked some random HTTP only website) and it works without the `:insecure t`.

So thanks again for asking! This can be closed.
